### PR TITLE
refactor!: Remove config options to use default value in 3.0

### PR DESCRIPTION
**Pull request by jinlinGuan**
related:
https://github.com/edgexfoundry/go-mod-bootstrap/pull/406 https://github.com/edgexfoundry/edgex-go/pull/4255

Signed-off-by: Ginny Guan <ginny@iotechsys.com>

**If your build fails** due to your commit message not passing the build checks, please review the guidelines here: https://github.com/edgexfoundry/edgex-compose/blob/main/.github/Contributing.md

## PR Checklist
Please check if your PR fulfills the following requirements:

- [ ] I am not introducing a breaking change (if you are, flag in conventional commit message with `BREAKING CHANGE:` describing the break)
- [ ] I have fully tested (add details below) this the new feature or bug fix (if not, why?)
- [ ] I have opened a PR for the related docs change (if not, why?)
  <link to docs PR>

### CODE REVIEW

File: compose-builder/.env [hunk 1]
```diff
@@ -52,4 +52,4 @@ NATS_VERSION=2.9-alpine
 EDGEX_USER=2002
 EDGEX_GROUP=2001
 
-DEFAULT_EDGEX_RUN_CMD_PARMS="-cp=consul.http://edgex-core-consul:8500 --registry --confdir=/res"
+DEFAULT_EDGEX_RUN_CMD_PARMS="-cp=consul.http://edgex-core-consul:8500 --registry --configDir=/res"
```
Comment (lenny-intel) [inline compose-builder/.env:55]: IMO we should drop the `--configDir=/res` just like was done in the Dockerfiles. Here and in all these other files unless the value will be something different than the default of  `./res` or`/res`

File: compose-builder/Makefile [hunk 2]
```diff
@@ -822,9 +822,9 @@ ifeq (taf-secty, $(filter taf-secty,$(ARGS)))
 	# taf has its special place holder from taf-device-services-mods and thus we need to keep it
 	# and extend security related things on top of it
 	ds_virtual_ext:= $(shell GEN_EXT_DIR="$(GEN_EXT_DIR)" ./gen_secure_compose_ext.sh device-virtual \
-		device-virtual device-virtual ' -cp=consul.http:\/\/edgex-core-consul:8500 --registry --confdir=CONF_DIR_PLACE_HOLDER')
+		device-virtual device-virtual ' -cp=consul.http:\/\/edgex-core-consul:8500 --registry --configDir=CONF_DIR_PLACE_HOLDER')
 	ds_modbus_ext:= $(shell GEN_EXT_DIR="$(GEN_EXT_DIR)" ./gen_secure_compose_ext.sh device-modbus \
-		device-modbus device-modbus ' -cp=consul.http:\/\/edgex-core-consul:8500 --registry --confdir=CONF_DIR_PLACE_HOLDER')
+		device-modbus device-modbus ' -cp=consul.http:\/\/edgex-core-consul:8500 --registry --configDir=CONF_DIR_PLACE_HOLDER')
```
Comment (lenny-intel) [inline compose-builder/Makefile:827]: Is there a reason we need a CONF_DIR_PLACE_HOLDER here? 
Use of `CONF_DIR_PLACE_HOLDER` should be renamed to `CONFIG_DIR_PLACE_HOLDER` here and I think other files.

**Review by lenny-intel — CHANGES_REQUESTED** [review]

**Comment by lenny-intel** [comment]
Please rename the Commit massage to be a `refactor` rather than a `fix`. 
`fix` should only be used for actual `bug fixes`. This is just one part of the larger refactoring.

**Comment by cloudxxx8** [comment]
Since the default value is `/res`, we agree to drop all the `--configDir` and `CONF_DIR_PLACE_HOLDER` related code.

File: compose-builder/Makefile (hunk 2 above)
Comment (jinlinGuan) [reply compose-builder/Makefile:827]: `CONF_DIR_PLACE_HOLDER` is replaced by a variable `${CONF_DIR}` in TAF. Will need to update [the TAF script](https://github.com/edgexfoundry/edgex-taf/blob/03ddac7a5263b1ef5c76871b10e0b56880d07258/TAF/utils/scripts/docker/get-compose-file.sh#L37) as well.

Comment (cloudxxx8) [reply compose-builder/Makefile:827]: I see.  We can keep those part for the TAF compose files.
@cherrycl for awareness

Comment (cloudxxx8) [reply compose-builder/Makefile:827]: The rename of `CONFIG_DIR_PLACE_HOLDER` will cause the TAF script change.
`${CONF_DIR}` in TAF should also be renamed to `${CONFIG_DIR}`

**Comment by cherrycl** [comment]
Smoke test got error `the format of Provider URL is incorrect (consul.http://edgex-core-consul:8500 --registry): parse \"consul.http://edgex-core-consul:8500 --registry\": invalid port \":8500 --registry\" after host` on all services which command contains `"-cp=consul.http://edgex-core-consul:8500 --registry"`.

**Comment by cherrycl** [comment]
All TAF compose file include no-secty files also needs to rename `CONF_DIR_PLACE_HOLDER` to `CONFIG_DIR_PLACE_HOLDER` and volume of device-service also contain `CONF_DIR_PLACE_HOLDER` string.

**Comment by cloudxxx8** [comment]
waiting for device services merged

**Review by lenny-intel — APPROVED** [review]
LGTM

**Comment by lenny-intel** [comment]
recheck

**Review by bnevis-i — DISMISSED** [review]

File: taf/docker-compose-taf-no-secty.yml [hunk 3]
```diff
@@ -403,7 +403,7 @@ services:
     - no-new-privileges:true
     user: 2002:2001
   device-virtual:
-    command: -cp=consul.http://edgex-core-consul:8500 --registry --confdir=CONF_DIR_PLACE_HOLDER
+    command: -cp=consul.http://edgex-core-consul:8500 --registry --confdir=CONFIG_DIR_PLACE_HOLDER
```
Comment (cherrycl) [inline taf/docker-compose-taf-no-secty.yml:406]: Should be updated to configdir for device-services?

Comment (lenny-intel) [reply taf/docker-compose-taf-no-secty.yml:406]: Yes

File: compose-builder/add-taf-device-services-mods.yml [hunk 4]
```diff
@@ -19,13 +19,13 @@ volumes:
 
 services:
   device-virtual:
-    command: "-cp=consul.http://edgex-core-consul:8500 --registry --confdir=CONF_DIR_PLACE_HOLDER"
+    command: "-cp=consul.http://edgex-core-consul:8500 --registry --confdir=CONFIG_DIR_PLACE_HOLDER"
```
Comment (lenny-intel) [inline compose-builder/add-taf-device-services-mods.yml:22]:
```suggestion
    command: "-cp=consul.http://edgex-core-consul:8500 --registry --configDir=CONFIG_DIR_PLACE_HOLDER"
```

File: compose-builder/add-taf-device-services-mods.yml [hunk 5]
```diff
@@ -19,13 +19,13 @@ volumes:
...
     volumes:
-      - PROFILE_VOLUME_PLACE_HOLDER:CONF_DIR_PLACE_HOLDER:z
+      - PROFILE_VOLUME_PLACE_HOLDER:CONFIG_DIR_PLACE_HOLDER:z
 
   device-modbus:
-    command: "-cp=consul.http://edgex-core-consul:8500 --registry --confdir=CONF_DIR_PLACE_HOLDER"
+    command: "-cp=consul.http://edgex-core-consul:8500 --registry --confdir=CONFIG_DIR_PLACE_HOLDER"
```
Comment (lenny-intel) [inline compose-builder/add-taf-device-services-mods.yml:27]:
```suggestion
    command: "-cp=consul.http://edgex-core-consul:8500 --registry --configDir=CONFIG_DIR_PLACE_HOLDER"
```

**Review by lenny-intel — CHANGES_REQUESTED** [review]

**Review by lenny-intel — APPROVED** [review]
LGTM

**Comment by cloudxxx8** [comment]
recheck

**Comment by cherrycl** [comment]
[edgex-taf PR #753](https://github.com/edgexfoundry/edgex-taf/pull/754) is required for this PR.

**Comment by cloudxxx8** [comment]
recheck

**Comment by cherrycl** [comment]
recheck

**Pull request merged**